src/custom/components/Tokens/TokensTable.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState, useMemo, useCallback, useEffect } from 'react'
+import { Token } from '@uniswap/sdk-core'
+import { ThemedText } from 'theme'
+import Loader from 'components/Loader'
+import LoadingRows from 'components/LoadingRows'
+import { AutoColumn } from 'components/Column'
+import TokensTableRow from './TokensTableRow'
+import { Label, Wrapper, TableHeader, TableBody, Break, PageButtons, Arrow, ArrowButton, ClickableText } from './styled'
+
+const MAX_ITEMS = 10
+
+const SORT_FIELD = {
+  name: 'name',
+}
+
+type TokenTableParams = {
+  tokensData: Token[] | undefined
+  maxItems?: number
+}
+
+export default function TokenTable({ tokensData, maxItems = MAX_ITEMS }: TokenTableParams) {
+  // sorting
+  const [sortField, setSortField] = useState(SORT_FIELD.name)
+  const [sortDirection, setSortDirection] = useState<boolean>(false)
+
+  // pagination
+  const [page, setPage] = useState(1)
+  const [maxPage, setMaxPage] = useState(1)
+
+  const sortedTokens = useMemo(() => {
+    return tokensData
+      ? tokensData
+          .filter((x) => !!x)
+          .sort((a, b) => {
+            const sortA = a[sortField as keyof Token]
+            const sortB = b[sortField as keyof Token]
+
+            if (!sortA || !sortB) return 0
+
+            if (a && b) {
+              return sortA > sortB ? (sortDirection ? -1 : 1) * 1 : (sortDirection ? -1 : 1) * -1
```

Comment:
Nice u even implemented sorting in different directions. 

I wonder what will be our sorting criteria if we have balances and favourites...  
I guess the order will be `favourite, balance, alphabetical` and if we invert `favourite, balance, alphabetical desc`

Reply:
Yea something like that, we can think about this at some future PR's